src/main/java/edu/harvard/iq/dataverse/DatasetPage.java

```diff
@@ -2109,8 +2117,15 @@ public void setSelectAllFiles(boolean selectAllFiles) {
         this.selectAllFiles = selectAllFiles;
     }
     
-    public void toggleSelectedFiles(){
-        //method for when user clicks (de-)select all files
+    public void toggleAllSelected(){
+        this.selectAllFiles = !this.selectAllFiles;
```

Comment:
At first blush `this.selectAllFiles = !this.selectAllFiles` was confusing to me but I understand that this is a toggle. Clicking means "reverse the state of this thing". We could add a comment to clarify this.